src/main/kotlin/net/maizegenetics/phgv2/cli/LoadVcf.kt

```diff
@@ -16,15 +32,158 @@ class LoadVcf : CliktCommand() {
+        // return a pair of lists, one for gvcf files and one for hvcf files
+        val vcfFiles = File(vcfDir).listFiles()
+        val gvcfFiles = mutableListOf<String>()
+        val hvcfFiles = mutableListOf<String>()
+        for (file in vcfFiles) {
+            if (file.name.endsWith("g.vcf.gz") || file.name.endsWith("gvcf.gz")) {
+                gvcfFiles.add(file.toString())
+            } else if (file.name.endsWith("h.vcf.gz") || file.name.endsWith("hvcf.gz")) {
+                hvcfFiles.add(file.toString())
+            }
+        }
+        return Pair(gvcfFiles,hvcfFiles)
+    }
+
+    // The uri should either be gvcf_dataset or hvcf_dataset
+    // The user determines the parent folder name where these datasets live
+    // The actual tiledb dataset names are constant and are either gvcf_dataset or hvcf_dataset
+
+    // We are only verifying, not creating the datasets
+    fun verifyURI(dbPath:String,uri:String): Boolean {
+        // Check that the user supplied db folder exists
+        check(File(dbPath).exists()) { "Folder $dbPath does not exist - please send a valid path that indicates the parent folder for your tiledb datasets." }
+
+        // Check if the dataset exists
+        val dataset = "${dbPath}/${uri}"
+        val datasetPath = Paths.get(dataset)
+
+        if (File(dataset).exists() && Files.isRegularFile(datasetPath)) {
+            throw IllegalArgumentException("URI ${dataset}is a file, not a tiledb dataset folder.  The parent folder must not contain any files/folders named gvcf_dataset or hvcf_dataset that is not a tiledb created URI")
+        }
+
+        if (File(dataset).exists()  && Files.isDirectory(Paths.get(dataset))){
+            // check if is a tiledb dataset
+            var builder = ProcessBuilder("conda","run","-n","phgv2-conda","tiledbvcf","stat","--uri",dataset)
+            var redirectOutput = tempDir + "/tiledb_statURI_output.log"
+            var redirectError = tempDir + "/tiledb_statURI_error.log"
+            builder.redirectOutput( File(redirectOutput))
+            builder.redirectError( File(redirectError))
+
+            // verify if the output.log contains "Version"
+            // if not, then the URI is not a tiledbvcf URI
+            println("begin Command:" + builder.command().stream().collect(Collectors.joining(" ")))
+
+            try {
+                var process = builder.start()
+                var error = process.waitFor()
+                if (error != 0) {
+                    println("LoadTiledbH tiledb stat returned error code $error")
+                    throw IllegalArgumentException("Error: URI is not a tiledb URI folder created via the tiledb create command: ${error}")
+                }
+
+            } catch (exc: Exception) {
+                println("Error: could not run tiledb stat on ${uri}.")
+                throw IllegalArgumentException("Error running ProcessBuilder to stat tiledb URI: ${exc.message}")
```

Comment:
I prefer including the exception (exc) as an additional argument to the thrown IllegalArgumentException so that the stack trace gets printed.